tests/templates/test_embeddings/test_amplitude.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the AmplitudeEmbedding template.
+"""
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane import numpy as pnp
+
+FEATURES = [
+    np.array([0, 1, 0, 0]),
+    1 / np.sqrt(4) * np.array([1, 1, 1, 1]),
+    np.array([complex(-np.sqrt(0.1), 0.0), np.sqrt(0.3), complex(0, -np.sqrt(0.1)), np.sqrt(0.5)]),
+]
+
+NOT_ENOUGH_FEATURES = [
+    np.array([0, 1, 0]),
+    1 / np.sqrt(3) * np.array([1, 1, 1]),
+    np.array([complex(-np.sqrt(0.1), 0.0), np.sqrt(0.3), complex(0, -np.sqrt(0.6))]),
+]
+
+TOO_MANY_FEATURES = [
+    [0, 0, 0, 1, 0],
+    1 / np.sqrt(8) * np.array([1] * 8),
+    [complex(-np.sqrt(0.1), 0.0), np.sqrt(0.3), complex(0, -np.sqrt(0.6)), 0.0, 0.0],
+]
+
+
+class TestDecomposition:
```

Comment:
Are these tests strictly for decomposition? e.g. `test_prepares_correct_state` doesn't seem about decomposition.

Reply:
Maybe there is a better name? In a sense, we are testing that the right gate sequence is used in the `expand` function...if not, we would not get the correct state